src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,161 @@
+            setItem(itemFromAPI);
+            setSharedLink(sharedLinkFromAPI);
+            setErrorMessage(null);
+        };
+
+        const getItem = () => {
+            const fields = {
+                fields: [
+                    FIELD_ALLOWED_INVITEE_ROLES,
+                    FIELD_DESCRIPTION,
+                    FIELD_EXTENSION,
+                    FIELD_ID,
+                    FIELD_NAME,
+                    FIELD_PERMISSIONS,
+                    FIELD_SHARED_LINK,
+                    FIELD_SHARED_LINK_FEATURES,
+                    FIELD_ITEM_TYPE,
+                ],
+            };
+            if (itemType === TYPE_FILE) {
+                api.getFileAPI().getFile(itemID, getItemSuccess, getError, fields);
+            } else if (itemType === TYPE_FOLDER) {
+                api.getFolderAPI().getFolderFields(itemID, getItemSuccess, getError, fields);
+            }
+        };
+
+        if (!item && !sharedLink) {
+            getItem();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    useEffect(() => {
+        const getUserSuccess = userData => {
+            const { id, userEnterpriseData } = convertUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorMessage(null);
+        };
+
+        const getUserData = () => {
+            api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
+                fields: [FIELD_ENTERPRISE, FIELD_HOSTNAME],
+            });
+        };
+        if (item && sharedLink && !currentUserID) {
+            getUserData();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    const renderElement = () => {
+        if (errorMessage) {
+            return <ErrorMask errorHeader={<FormattedMessage {...errorMessage} />} />;
+        }
+
+        if (item && sharedLink) {
+            return (
+                <Internationalize language={language} messages={messages}>
+                    <UnifiedShareModal
+                        canInvite={sharedLink.canInvite}
+                        changeSharedLinkPermissionLevel={() => Promise.resolve([])} // to do: replace with a POST to the Shared Link API
+                        collaboratorsList={{ collaborators: [] }} // to do: replace with Collaborators API
+                        currentUserID={currentUserID}
+                        getCollaboratorContacts={() => Promise.resolve([])} // to do: replace with Collaborators API
+                        getSharedLinkContacts={() => Promise.resolve([])} // to do: replace with Collaborators API
+                        initialDataReceived
+                        item={item}
+                        onAddLink={() => Promise.resolve([])} // to do: replace with a POST to the Shared Link API
+                        sharedLink={sharedLink}
+                        showFormOnly
```

Comment:
Should this be passing in `props.showFormOnly`? Also, I don't know if this is a super intuitive sounding name to me... but maybe just me? Does it make sense to opt out of the modal, or opt-in? Which is the most common use case? Maybe something like `displayInModal` or `renderInModal` (it could default to true if we'd rather do opt-out)...?

Reply:
I believe the modal is the more common use case. `displayInModal` sounds good to me; I updated the code with the new prop.